pkg/util/system/network_linux.go

```diff
@@ -0,0 +1,158 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+// +build linux
+
+package system
+
+import (
+	"fmt"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/filesystem"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// NetworkRoute holds one network destination subnet and it's linked interface name
+type NetworkRoute struct {
+	Interface string
+	Subnet    uint64
+	Gateway   uint64
+	Mask      uint64
+}
+
+// ParseProcessRoutes parses /proc/<pid>/net/route into a list of NetworkDestionation
+// If PID is 0, it parses /proc/net/route instead
+func ParseProcessRoutes(procPath string, pid int) ([]NetworkRoute, error) {
+	var procNetFile string
+	if pid > 0 {
+		procNetFile = filepath.Join(procPath, strconv.Itoa(pid), "net", "route")
+	} else {
+		procNetFile = filepath.Join(procPath, "net", "route")
+	}
+
+	if !filesystem.FileExists(procNetFile) {
+		return nil, fmt.Errorf("%s not found", procNetFile)
+	}
+	lines, err := filesystem.ReadLines(procNetFile)
+	if err != nil {
+		return nil, err
```

Comment:
Would it make sense to decorate the error with the failed path here as well, like other errors in this func?